craft_application/models/manifest.py

```diff
@@ -14,19 +14,35 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """Models representing manifests for projects and fetch-service assets."""
+import hashlib
+import pathlib
+from datetime import datetime, timezone
 from typing import Any, Literal
 
 from pydantic import Field
 from typing_extensions import override
 
+from craft_application import models
 from craft_application.models import CraftBaseModel
 
 
 class Hashes(CraftBaseModel):
     """Digests identifying an artifact/asset."""
 
-    sha1: str = Field(alias="SHA1")
-    sha256: str = Field(alias="SHA256")
+    sha1: str
+    sha256: str
+
+    @staticmethod
+    def from_path(path: pathlib.Path) -> "Hashes":
+        """Compute digests for a given path."""
+        read_bytes = path.read_bytes()
+
+        return Hashes(
+            sha1=hashlib.sha1(  # noqa: S324 (insecure hash function)
+                read_bytes
+            ).hexdigest(),
+            sha256=hashlib.sha256(read_bytes).hexdigest(),
+        )
```

Comment:
Since this is an alternative constructor, how about

```suggestion
    @classmethod
    def from_path(cls, path: pathlib.Path) -> Self:
        """Compute digests for a given path."""
        read_bytes = path.read_bytes()

        return cls(
            sha1=hashlib.sha1(  # noqa: S324 (insecure hash function)
                read_bytes
            ).hexdigest(),
            sha256=hashlib.sha256(read_bytes).hexdigest(),
        )
```